ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java

```diff
@@ -326,82 +282,46 @@ void onAdEnd(String placement, boolean wasSuccessfulView, boolean wasCallToActio
             }
         }
 
+        @Override
+        void onAdAvailable() {
+            if (mMediationBannerListener != null) {
+                mMediationBannerListener.onAdLoaded(VungleInterstitialAdapter.this);
+            }
+        }
+
         @Override
         void onAdStart(String placement) {
             //let's load it again to mimic auto-cache, don't care about errors
-            mVungleManager.loadAd(placement, mAdConfig.getAdSize(), null);
+            if (mBannerRequest != null) {
+                mBannerRequest.preCache();
+            }
         }
 
         @Override
         void onAdFail(String placement) {
-            Log.w(TAG, "Ad playback error Placement: " + placement);
+            Log.w(TAG, "Ad playback error Placement: " + placement + ";" + mBannerRequest);
             if (mMediationBannerListener != null) {
                 mMediationBannerListener.onAdFailedToLoad(VungleInterstitialAdapter.this,
                         AdRequest.ERROR_CODE_INTERNAL_ERROR);
             }
         }
-    };
-
-    private void createBanner() {
-        if (mVungleManager == null || !pendingRequestBanner.get())
-            return;
 
-        mVungleManager.cleanUpBanner(mPlacementForPlay);

```

Comment:
with onDestroy() missing sometimes can't stop removing any banner for mPlacementForPlay

Reply:
yes, I can reproduce it:
test case: press 'show banners' button in DefaultActivity, when adView is loading and not display then back press asap(must make sure adView not showing).
root cause: when back pressed adView#destroy fired, but onDestroy not called on adapter side, meanwhile, the ad request still running and put it to the map. So we will miss remove it.

Reply:
same as for previous test app code: when we destroy adView, remove it from parent and creating a new one.